### DISCUSSION TITLE
Pairwise analysis for beta and alpha diversities

### DISCUSSION THREAD

**Discussion by aortizchura** [cafferychen777/MicrobiomeStat#25] [Q&A] [answered]
HI,
Thank you for this tool, which I find very useful. I was wondering if we can perform the pairwise comparison for alpha and beta diversities?

Abimael

**Comment by cafferychen777** [comment, marked as answer]
Hello Abimael,

Thank you for your interest in the MicrobiomeStat package and for your question regarding the analysis of alpha and beta diversities.

Yes, MicrobiomeStat does support pairwise comparisons for both alpha and beta diversities. Here's a brief overview of how you can perform these analyses:

1. **Alpha Diversity Analysis**: For alpha diversity, you can explore different metrics and perform analyses tailored to your study's needs. Our package provides various functions to calculate alpha diversity and compare it across different groups. More detailed information and step-by-step instructions are available at our [alpha diversity analysis page](https://www.microbiomestat.wiki/paired-samples-analysis/unraveling-paired-samples-alpha-diversity-analysis).

2. **Beta Diversity Analysis**: Similarly, for beta diversity, MicrobiomeStat offers a range of tools to examine the variations in microbial community structure across different groups and within individual subjects over time. The package supports several distance measures like Bray-Curtis, Jaccard, and UniFrac, among others. You can perform statistical testing of beta diversity differences, generate ordination plots, and create various visualizations to better understand your data. For more information on beta diversity analysis, please refer to our [beta diversity analysis page](https://www.microbiomestat.wiki/paired-samples-analysis/navigating-paired-samples-beta-diversity-analysis).

These resources should guide you through the process of performing pairwise comparisons in your microbial community analyses. If you have any more specific questions or need further assistance, feel free to reach out!

Best regards,
Chen YANG

**Comment by antonija-kolobaric** [comment]
Hi Chen -- thank you for this amazing tool. I have a follow-up question on this topic:

I am working with a dataset where we collected two microbial samples from the same subjects using two different methods. In other words, my dataset has 20 subjects (20 different IDNo), and each subject provided 2 samples collected using two different methods, giving me 40 samples overall. I want to compare if the methods produce different results. I used the paired samples analysis wiki to do the analysis.

When analyzing the data, I ran the following analysis for alpha diversity 

alpha_div <- generate_alpha_test_pair(
  data.obj = MicrobiomeDataR,
  alpha.obj = NULL,
  time.var = "Method",
  alpha.name = c("shannon", "simpson", "chao1"),
  subject.var = "IDNo",
  group.var = NULL
)

If I add "Method" as group variable, I lose the dashed lines connecting the same subject when plotting alpha diversities, so I set group to NULL.
I tried running beta diversity with the same setup as well as with setting group="Method":

beta_div <- generate_beta_change_test_pair(
  data.obj = MicrobiomeData,
  dist.obj = NULL,
  time.var = “Method”,
  subject.var = “IDNo”,
  group.var = “Method”,
  adj.vars = c(“batch”),
  change.base = “1”,
  dist.name = c(‘BC’, ‘Jaccard’)
)

In both cases I get an error error: "contrasts can be applied only to factors with 2 or more levels".

What would be the correct way to set up my analysis? I have paired samples, so I am assuming I should do paired sample analysis. However, I am unsure about the time/group variables. Any help would be appreciated!

**Reply by cafferychen777** [reply]
Dear Antonija,

Thank you for your question and interest in the MicrobiomeStat package. From what I understand, you have a cross-sectional dataset with 20 subjects, each providing two samples collected using different methods. Your goal is to compare the methods and determine if they produce different results.

In this case, I recommend treating your data as a cross-sectional dataset rather than a paired dataset. Although the samples come from the same subjects, they are collected using different methods at a single time point, which aligns with the definition of a [cross-sectional study](https://www.microbiomestat.wiki/single-point-analysis/unraveling-cross-sectional-studies-with-microbiomestat).

To analyze your data, please follow the instructions in the "Unraveling Cross-Sectional Studies with MicrobiomeStat" section of our wiki: https://www.microbiomestat.wiki/single-point-analysis/unraveling-cross-sectional-studies-with-microbiomestat

For alpha diversity analysis, use the `generate_alpha_test_single` function:

```R
alpha_div <- generate_alpha_test_single(
  data.obj = MicrobiomeDataR,
  alpha.obj = NULL,
  group.var = "Method",
  alpha.name = c("shannon", "simpson", "chao1")
)
```

For beta diversity analysis, use the `generate_beta_test_single` function:

```R
beta_div <- generate_beta_test_single(
  data.obj = MicrobiomeData,
  dist.obj = NULL,
  group.var = "Method",
  adj.vars = c("batch"),
  dist.name = c('BC', 'Jaccard')
)
```

By using the `group.var = "Method"` argument, you will be able to compare the two sample collection methods and determine if they produce significantly different results in terms of alpha and beta diversities.

I hope this clarifies the appropriate approach for your analysis. If you have any further questions, please don't hesitate to ask.

Best regards,
Chen

**Reply by antonija-kolobaric** [reply]
Hi Chen,

Thank you for a quick response! I should have noted this, but the samples were collected at two different time points! The first sample was collected at the time of the diagnosis, and the second sample was collected a few weeks/months later (time varies for each participant) during surgery.

**Reply by cafferychen777** [reply]
Hi Antonija,

Thank you for providing more details about your dataset. Since your samples were collected at two different time points (diagnosis and surgery) for each subject, you can indeed use the paired analysis functions in MicrobiomeStat.

For the alpha diversity analysis, you can use the code you provided earlier:

alpha_div <- generate_alpha_test_pair(
  data.obj = MicrobiomeDataR,
  alpha.obj = NULL,
  time.var = "Method",
  alpha.name = c("shannon", "simpson", "chao1"),
  subject.var = "IDNo",
  group.var = NULL
)

Here, the "Method" variable represents the two time points (diagnosis and surgery) for each subject. By setting group.var = NULL, you ensure that the dashed lines connecting the same subject are displayed in the alpha diversity plots.

Please note that some paired analysis functions in MicrobiomeStat may not work due to statistical limitations. However, I believe the remaining functions should be sufficient to address your research questions.

If you encounter any issues or have further questions, please don't hesitate to reach out.

Best regards,
Chen
